executor/tests/test_executor.py

```diff
@@ -306,12 +314,7 @@ def test_await_process_completion_killed(self):
             shutdown_grace_period_ms = 2000
 
             stop_signal = Event()
-
-            def sleep_and_set_stop_signal():
-                time.sleep(2 * cook.RUNNING_POLL_INTERVAL_SECS)
-                stop_signal.set()
-            thread = Thread(target=sleep_and_set_stop_signal, args=())
-            thread.start()
+            Thread(target=sleep_and_set_stop_signal, args=(stop_signal, 2 * cook.RUNNING_POLL_INTERVAL_SECS)).start()
```

Comment:
Is there any reason not to use a `Timer` here? You should be able to set it as a daemon thread to avoid holding up the whole process if it wants to exit before this timer goes off.

Reply:
See comment above.